bundles/org.openhab.core.model.yaml/src/main/java/org/openhab/core/model/yaml/internal/rules/YamlRuleTemplateProvider.java

```diff
@@ -0,0 +1,172 @@
+        return version >= 1;
+    }
+
+    @Override
+    public boolean isDeprecated() {
+        return false;
+    }
+
+    @Override
+    public void addedModel(String modelName, Collection<YamlRuleTemplateDTO> elements) {
+        List<RuleTemplate> added = elements.stream().map(this::mapRuleTemplate).filter(Objects::nonNull).toList();
+        Collection<RuleTemplate> modelRuleTemplates = Objects
+                .requireNonNull(ruleTemplatesMap.computeIfAbsent(modelName, k -> new ArrayList<>()));
+        modelRuleTemplates.addAll(added);
+        added.forEach(t -> {
+            logger.debug("model {} added rule template {}", modelName, t.getUID());
+            notifyListenersAboutAddedElement(t);
+        });
+    }
+
+    @Override
+    public void updatedModel(String modelName, Collection<YamlRuleTemplateDTO> elements) {
+        List<RuleTemplate> updated = elements.stream().map(this::mapRuleTemplate).filter(Objects::nonNull).toList();
+        Collection<RuleTemplate> modelRules = Objects
+                .requireNonNull(ruleTemplatesMap.computeIfAbsent(modelName, k -> new ArrayList<>()));
+        updated.forEach(t -> {
+            modelRules.stream().filter(template -> template.getUID().equals(t.getUID())).findFirst()
+                    .ifPresentOrElse(oldTemplate -> {
+                        modelRules.remove(oldTemplate);
+                        modelRules.add(t);
+                        logger.debug("model {} updated rule template {}", modelName, t.getUID());
+                        notifyListenersAboutUpdatedElement(oldTemplate, t);
+                    }, () -> {
+                        modelRules.add(t);
+                        logger.debug("model {} added rule template {}", modelName, t.getUID());
+                        notifyListenersAboutAddedElement(t);
+                    });
+        });
+    }
+
+    @Override
+    public void removedModel(String modelName, Collection<YamlRuleTemplateDTO> elements) {
+        List<RuleTemplate> removed = elements.stream().map(this::mapRuleTemplate).filter(Objects::nonNull).toList();
+        Collection<RuleTemplate> modelRuleTemplates = ruleTemplatesMap.getOrDefault(modelName, List.of());
+        removed.forEach(t -> {
+            modelRuleTemplates.stream().filter(template -> template.getUID().equals(t.getUID())).findFirst()
+                    .ifPresentOrElse(oldTemplate -> {
+                        modelRuleTemplates.remove(oldTemplate);
+                        logger.debug("model {} removed rule template {}", modelName, t.getUID());
+                        notifyListenersAboutRemovedElement(oldTemplate);
+                    }, () -> logger.debug("model {} rule template {} not found", modelName, t.getUID()));
+        });
+        if (modelRuleTemplates.isEmpty()) {
+            ruleTemplatesMap.remove(modelName);
+        }
+    }
+
+    private @Nullable RuleTemplate mapRuleTemplate(YamlRuleTemplateDTO ruleTemplateDto) {
```

Comment:
I don't see the case the method will return null.
If you changed here to remove `@Nullable`, of course you have to adjust where the method is called.

Reply:
It now returns null if the parsing fails, it was the problem with the wrong exception type that caused this whole mess, because the `SerializationException` that was used by mistake was unchecked.